### ISSUE TITLE
Comments on similarities with SNN 

### ISSUE THREAD

**Issue by parashardhapola** [lmcinnes/umap#63]
Hi @lmcinnes,

I've been trying out UMAP on some single cell RNA-Seq dataset. I'm pretty surprised (in a pleasing way) by the similarity between UMAP transformed dimensions and a force directed layout of the shared nearest neighbor (SNN) graph of the same data. Obviously UMAP was blazing fast in comparison. After trying this on multiple data sets, I see that SNN graph topologies are consistenly similar to UMAP embedding.

![2018-04-30-181945_1920x1080_scrot](https://user-images.githubusercontent.com/10581156/39438105-40f9fa32-4ca3-11e8-9f14-b6213b665623.png)

I'm hoping that you could comment on why the SNN graph topologies are this similar to UMAP's embedding.

Given these empirical similarities between the two methods, I'm lured to use UMAP embedding as seed positions to calculate the graph layout (and cut down the iterations for layout calculations). However, I'm not sure if this would make any sense. Do you see any issue with this approach?

Thanks.

**Comment by lmcinnes** [comment]
I am familiar with ForceAtlas in general, but don't know the internals of the algorithm very well. None the less I can say that the intermediate topological representation that UMAP uses can be viewed as a weighted shared neighbor graph, and that the layout optimization can be broken down as attractive and repulsive forces applied according to that graph, so I would expect some similarities. On the other hand t-SNE can also be described in those same broad terms, so depending on how t-SNE compares I think you should get some idea of how well that analogy holds up.

In short -- I would expect some level of similarity, but I can't really say how deep it goes. Sorry I can't be more help at this time. If I get some spare time to read about ForceAtlas in more depth perhaps I can say more.

**Comment by parashardhapola** [comment]
Thanks for your reply. The fact that UMAP uses a SNN graph during intermediate steps might explain (atleast partially) the similarities. The graph's topologies look quite similar if I was to use Fruchterman-Reingold force-directed algorithm instead of ForceAtlas2 (but it runs painfully slow). The TSNE also qualitatively resembles the graph topology but the embedding feels more noisy (or more spread out) and is quite sensitive to the value for parameter `perplexity`. 

<img src="https://user-images.githubusercontent.com/10581156/39442772-a1f490a4-4cb2-11e8-84cb-12a02b1a9b10.png" width="400" />

This behavior of TSNE vis-a-vis UMAP can also be observed in this [preprint](https://www.biorxiv.org/content/early/2018/04/10/298430), wherein similar samples (data points) are 'squeezed' together by UMAP, while they seem more spread out on TSNE.

Anyway, given UMAP's fast speed and similarity to vanilla weighted SNN, I'm inclined to use it for seeding graph layout calculation and see how it works (unless a flaw can be seen in this approach)

Thank you again for your insight.

**Comment by lmcinnes** [comment]
The extra spreading (and sensitivity to perplexity) in t-SNE is what I would expect - -the repulsion force is a little different and should produce results like that. Given all of that I think that yes, this would be a suitable approach to initially seeding Force Atlas computations.
